### ISSUE TITLE
Exception in Xaml Preview in Visual Studio for Mac

### ISSUE THREAD

**Issue by follesoe** [MvvmCross/MvvmCross#3031]
## 🐛 Bug Report

I'm receiving an error in the Xaml Preview tool in Visual Studio for Mac when opening a Xaml file for a MvxContentPage containing a Mvx binding: `mvx:Bi.nd="Text Message"`.

Error: `The type initializer for 'MvvmCross.Forms.Bindings.Bi' threw an exception`

### Expected behavior
Preview to function as for any Xamarin Forms Page.

### Reproduction steps
Open a MvvmCross Xamarin Forms Xaml Page containing MvvmCross binding properties in Visual Studio for Mac.

Example XAML:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<d:MvxContentPage
    x:Class="MyApp.Views.MyPage"
    x:TypeArguments="viewModels:MyViewModel"    
    xmlns="http://xamarin.com/schemas/2014/forms"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:mvx="clr-namespace:MvvmCross.Forms.Bindings;assembly=MvvmCross.Forms"    
    xmlns:viewModels="clr-namespace:MyApp.ViewModels;assembly=MyApp"
    xmlns:d="clr-namespace:MvvmCross.Forms.Views;assembly=MvvmCross.Forms">
    <d:MvxContentPage.Content>
        <StackLayout VerticalOptions="Center">
            <Label
                mvx:Bi.nd="Text Message"
                VerticalOptions="Center" 
                HorizontalOptions="Center" />
        </StackLayout>
    </d:MvxContentPage.Content>
</d:MvxContentPage>
```

### Configuration

**Version:** 6.1.2

**Platform:** 
- [ ] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [x] :monkey: Xamarin.Forms

**Comment by follesoe** [comment]
I know this could be a Visual Studio bug, but at the same time, as `mvx:Bi.nd` is the recommended way to express bindings in Xaml, it would be nice if the feature worked within the limitations of the current tools. Perhaps I'm missing something in my class lib to ensure the preview tool is able to initialize the `Bi` type, but so far I have not been able to understand why this is happening. If I revert to traditional Xamarin.Forms `{Binding Path=...}` expressions the preview tool works.

**Comment by Cheesebaron** [comment]
There must be more to the stack trace? If not directly in the UI, could you please provide logs? You should be able to find them under the help menu in VSfM

**Comment by nickrandolph** [comment]
@Cheesebaron @martijn00 not sure if this is the issue but we should change this line https://github.com/MvvmCross/MvvmCross/blob/8fbc13caa44e08f3fe2eb3732f099ad534803b43/MvvmCross.Forms/Bindings/MvxDesignTimeChecker.cs#L25
I think we should make use of the DesignMode class: https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.designmode?view=xamarin-forms

**Comment by follesoe** [comment]
Could not find an easy way to copy the stack trace from the error dialog in the XAML preview window. This is a screenshot of the stack trace:

<img width="640" alt="screen shot 2018-07-25 at 00 36 46" src="https://user-images.githubusercontent.com/169999/43169829-d6984548-8fa2-11e8-9883-14e79ed2120a.png">
<img width="639" alt="screen shot 2018-07-25 at 00 36 38" src="https://user-images.githubusercontent.com/169999/43169833-da1c6938-8fa2-11e8-8827-964283485a4b.png">

**Comment by follesoe** [comment]
So no MvvmCross name spaces on the stack trace, so must be something that goes wrong when it starts to access the `Bi`-class. So probably reproducable through code/unit testing via the `LoadFromXaml` method?

**Comment by follesoe** [comment]
Found the log:

```
2018-07-25 00:35:17.0] INFO: Reloading: Server rendering completed in 15ms
[2018-07-25 00:35:17.0] ERROR: : iOSXamlRenderer had a rendering error: System.TypeInitializationException: The type initializer for 'MvvmCross.Forms.Bindings.Bi' threw an exception.
<unknown method>
System.Reflection.MonoField.GetValue (System.Object obj) in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.12.0.4/src/Xamarin.iOS/mcs/class/corlib/System.Reflection/MonoField.cs:243
Xamarin.Forms.Xaml.ApplyPropertiesVisitor.GetBindableProperty (System.Type elementType, System.String localName, System.Xml.IXmlLineInfo lineInfo, System.Boolean throwOnError) in D:\a\1\s\Xamarin.Forms.Xaml\ApplyPropertiesVisitor.cs:285
Xamarin.Forms.Xaml.ApplyPropertiesVisitor.SetPropertyValue (System.Object xamlelement, Xamarin.Forms.Xaml.XmlName propertyName, System.Object value, System.Object rootElement, Xamarin.Forms.Xaml.INode node, Xamarin.Forms.Xaml.HydrationContext context, System.Xml.IXmlLineInfo lineInfo) in D:\a\1\s\Xamarin.Forms.Xaml\ApplyPropertiesVisitor.cs:317
Xamarin.Forms.Xaml.ApplyPropertiesVisitor.Visit (Xamarin.Forms.Xaml.ValueNode node, Xamarin.Forms.Xaml.INode parentNode) in D:\a\1\s\Xamarin.Forms.Xaml\ApplyPropertiesVisitor.cs:57
Xamarin.Forms.Xaml.ValueNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:86
Xamarin.Forms.Xaml.ElementNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:143
Xamarin.Forms.Xaml.ElementNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:145
Xamarin.Forms.Xaml.RootNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:200
Xamarin.Forms.Xaml.XamlLoader.Visit (Xamarin.Forms.Xaml.RootNode rootnode, Xamarin.Forms.Xaml.HydrationContext visitorContext) in D:\a\1\s\Xamarin.Forms.Xaml\XamlLoader.cs:143
Xamarin.Forms.Xaml.XamlLoader.Load (System.Object view, System.String xaml) in D:\a\1\s\Xamarin.Forms.Xaml\XamlLoader.cs:89
Xamarin.Forms.Xaml.XamlLoader.Load (System.Object view, System.Type callingType) in D:\a\1\s\Xamarin.Forms.Xaml\XamlLoader.cs:67
Xamarin.Forms.Xaml.Extensions.LoadFromXaml[TXaml] (TXaml view, System.Type callingType) in D:\a\1\s\Xamarin.Forms.Xaml\ViewExtensions.cs:36
Blueye.App.Views.SettingsPage.__InitComponentRuntime ()
Blueye.App.Views.SettingsPage.InitializeComponent () in /Users/follesoe/Source/BlueyeApp/Core/obj/Debug/netstandard2.0/Views/SettingsPage.xaml.g.cs:22
Blueye.App.Views.SettingsPage..ctor () in /Users/follesoe/Source/BlueyeApp/Core/Views/SettingsPage.xaml.cs:13
<unknown method>
System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.12.0.4/src/Xamarin.iOS/mcs/class/corlib/System.Reflection/MonoMethod.cs:661
```

**Comment by martijn00** [comment]
I know there was an error in the binding there that i fixed in 6.2.0-beta2, but not sure if it is related. Could you test?

**Comment by kharabyasser** [comment]
More than 8 hours search now more than 5 issues about this (since 2017) non has been resolved and closed!
